### ISSUE TITLE
Tesla Fleet API, can't add my car (dropdown empty)

### ISSUE THREAD

**Issue by josefzahner** [bassmaster187/TeslaLogger#1288]
When I try to add my vehicle via Tesla Fleet API I get an exception on step 4, so my dropdown menu which should contain the VIN is empty... any ideas what could be the issue?

<img width="1294" alt="Screenshot 2024-05-23 at 00 24 04" src="https://github.com/bassmaster187/TeslaLogger/assets/33521474/4e496f3b-2830-4d2d-ab23-c7f629389c4b">

Log:
```23.05.2024 00:24:15 : #-1[Thread Pool Worker:68]: FleetApiAddress: https://fleet-api.prd.eu.vn.cloud.tesla.com/
23.05.2024 00:24:15 : #-1[Thread Pool Worker:68]: Update fleetAPIaddress / Value: https://fleet-api.prd.eu.vn.cloud.tesla.com/
23.05.2024 00:24:15 : #-1[Thread Pool Worker:68]: FleetApiAddress: https://fleet-api.prd.eu.vn.cloud.tesla.com/
23.05.2024 00:24:15 : #-1[Thread Pool Worker:68]: Update fleetAPIaddress / Value: https://fleet-api.prd.eu.vn.cloud.tesla.com/
23.05.2024 00:24:26 : #-1[Thread Pool Worker:62]: Timeout: vehicle_data?endpoints=vehicle_state&let_sleep=true
23.05.2024 00:24:26 : System.AggregateException: One or more errors occurred. (A task was canceled.) ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
--- End of inner exception stack trace ---
at System.Threading.Tasks.Task.ThrowIfExceptional (System.Boolean includeTaskCanceledExceptions) [0x00011] in <533173d24dae460899d2b10975534bb0>:0
at System.Threading.Tasks.Task`1[TResult].GetResultCore (System.Boolean waitCompletionNotification) [0x0002b] in <533173d24dae460899d2b10975534bb0>:0
at System.Threading.Tasks.Task`1[TResult].get_Result () [0x0000f] in <533173d24dae460899d2b10975534bb0>:0
at TeslaLogger.WebHelper.DoGetVehiclesRequest (System.String& resultContent, System.Net.Http.HttpClient client, System.String adresse, System.Threading.Tasks.Task`1[System.Net.Http.HttpResponseMessage]& resultTask, System.Net.Http.HttpResponseMessage& result) [0x00075] in :0
at TeslaLogger.WebHelper.GetAllVehicles (System.String& resultContent, Newtonsoft.Json.Linq.JArray& vehicles, System.Boolean throwExceptionOnUnauthorized, System.Boolean doNotCache) [0x000f4] in :0
at TeslaLogger.WebServer.Admin_GetCarsFromAccount (System.Net.HttpListenerRequest request, System.Net.HttpListenerResponse response, System.Boolean fleetAPI) [0x0013d] in :0
---> (Inner Exception #0) System.Threading.Tasks.TaskCanceledException: A task was canceled.<---```

**Comment by bassmaster187** [comment]
What Model do you have?

**Comment by josefzahner** [comment]
Model Y Performance from June 2023

**Comment by josefzahner** [comment]
Btw. git clone from yesterday and running on docker on my Synology NAS

**Comment by bassmaster187** [comment]
Could you please retry getting a new token to make sure it wasn't a server problem.

**Comment by josefzahner** [comment]
Just checked, problem persists.... Token generation works, but no VIN available.

**Comment by josefzahner** [comment]
I've tried it again on fresh installed Raspberry Pi 4, but of course again with the Docker setup. Exactly the same behaviour as on my Synology. Dropdown empty :-(

**Comment by josefzahner** [comment]
Adding the car by "In local browser" works, dropdown with VIN filled there. However I don't know how long this will work or if it does (haven't driven until now)

**Comment by jaimebennett** [comment]
I have this same issue with the newest build.  Can I add the vehicle manually to the DB?

**Comment by andrelx** [comment]
Same issue here (or so I think). I have sold my previous Tesla and it has been removed from my Tesla account.  I now have only the new one on my Tesla account. When I authenticate, it comes back with an empty list for cars.

![Capture d’écran 2024-07-28 130931](https://github.com/user-attachments/assets/97389726-7914-4c4c-a5a8-2fb78acbfc69)

**Comment by bassmaster187** [comment]
could you please send me your logfile

**Comment by andrelx** [comment]
Just to be sure : is there any sensitive information in the logfile ? I wouldn't want to post publicly something thqat would give access to my car ;-)

**Comment by bassmaster187** [comment]
No, everything sensitive is obfuscated. You can send me the logfile by mail.

**Comment by andrelx** [comment]
I’ve sent you an email to ***@***.***', but not being sure, I resend it by simply replying.

Hi!

Attache is my Teslalogger logfile.  Let me give you a little more background.   I’m running Teslalogger on a Qnap using Container station / Docker.

On July 26th, I exchanged my Tesla Model 3 LR AWD 2018 for a new 2024 Model 3 Highland LR AWD. And I love it 😉

Obviously, since then, Teslalogger can’t access my old M3 as it is not on my Tesla account anymore.  What I’m trying to do is to add my new car to Teslalogger without deleting the old one, preventing the loss of the data I’ve collected for the last 5 ½ year with my previous car.

A couple of questions:

  *   Is this doable ?
  *   Can I name my new car the same name as the old one, and will that cause problem in teslalogger (besides having to select the proper “MrCar” in the dropdowns to get the data from the proper car) ?  In fact, can I change the car name aythime without disturbing Teslalogger ?
  *   And the main reason I’ve wrote a ticket : could you correct the app for the authentication not to return an empty list, but showing my current car (which is presently called “Model 3” by default)

Thank you very much for you fast support – it’s impressive !

André



De : Christian P. ***@***.***>
Envoyé : 28 juillet 2024 17:47
À : bassmaster187/TeslaLogger ***@***.***>
Cc : André Laurendeau ***@***.***>; Comment ***@***.***>
Objet : Re: [bassmaster187/TeslaLogger] Tesla Fleet API, can't add my car (dropdown empty) (Issue #1288)


No, everything sensitive is obfuscated. You can send me the logfile by mail.

**Comment by andrelx** [comment]
Any development of this ?

Thanks,

André
